app/scripts/modules/amazon/src/serverGroup/details/sections/CapacityDetailsSection.tsx

```diff
@@ -1,52 +1,50 @@
 import * as React from 'react';
 import { BindAll } from 'lodash-decorators';
 
-import { CollapsibleSection, NgReact, ReactInjector } from '@spinnaker/core';
+import { CollapsibleSection, NgReact, Overridable, ReactInjector, Application, IServerGroup } from '@spinnaker/core';
 
 import { IAmazonServerGroupDetailsSectionProps } from './IAmazonServerGroupDetailsSectionProps';
 
+@Overridable('amazon.serverGroup.CapacityDetailsSection')
 @BindAll()
 export class CapacityDetailsSection extends React.Component<IAmazonServerGroupDetailsSectionProps> {
-  private resizeServerGroup(): void {
+  public static resizeServerGroup(serverGroup: IServerGroup, application: Application): void {
     ReactInjector.modalService.open({
       templateUrl: ReactInjector.overrideRegistry.getTemplate('aws.resize.modal', require('../resize/resizeServerGroup.html')),
       controller: 'awsResizeServerGroupCtrl as ctrl',
       resolve: {
-        serverGroup: () => this.props.serverGroup,
-        application: () => this.props.app
+        serverGroup: () => serverGroup,
+        application: () => application,
       }
     });
   }
 
   public render(): JSX.Element {
-    const { serverGroup } = this.props;
+    const { serverGroup, app } = this.props;
     const { ViewScalingActivitiesLink } = NgReact;
+    const simple = serverGroup.asg.minSize === serverGroup.asg.maxSize;
 
     return (
       <CollapsibleSection heading="Capacity" defaultExpanded={true}>
-        {serverGroup.asg.minSize === serverGroup.asg.maxSize && (
-          <dl className="dl-horizontal dl-flex">
-            <dt>Min/Max</dt>
-            <dd>{serverGroup.asg.desiredCapacity}</dd>
-            <dt>Current</dt>
-            <dd>{serverGroup.instances.length}</dd>
-          </dl>
-        )}
-        {serverGroup.asg.minSize !== serverGroup.asg.maxSize && (
-          <dl className="dl-horizontal dl-flex">
-            <dt>Min</dt>
-            <dd>{serverGroup.asg.minSize}</dd>
-            <dt>Desired</dt>
-            <dd>{serverGroup.asg.desiredCapacity}</dd>
-            <dt>Max</dt>
-            <dd>{serverGroup.asg.maxSize}</dd>
-            <dt>Current</dt>
-            <dd>{serverGroup.instances.length}</dd>
-          </dl>
-        )}
+        <dl className="dl-horizontal dl-flex">
```

Comment:
I refactored this to be closer to the style in upcoming deck-nflx PR.

I don't love the `{boolean && <element/>}` thing, but this can be fixed with react 16 is live

Reply:
fragments ftw